pc2cps

```diff
@@ -72,18 +86,60 @@ class pkgconfig(object):
             else:
                 deps[dep[0]] = {}
 
-        return deps
+        self.requires = deps
 
     #--------------------------------------------------------------------------
     @property
     def component(self):
         def _keys(x):
-            return None if x is None else x.keys()
-
-        c = {
-            'Type': 'interface',
-        }
-        extend(c, 'Link-Flags', self._get_split('Libs'))
+            return [] if x is None else x.keys()
+
+        def _list(re_):
+            ma = [re_.match(x) for x in link_flags]
+            return [x.group(1) for x in ma if x]
+
+        # If not found and if package name given in library list with
+        # "-l" flag, an error should be thrown?
+        def _find_location():
+            for path in self._paths:
+                for file in os.listdir(path):
+                    pattern = '(lib)?' + self.library_name + self._lib_pattern
+                    ma = re.match(pattern, file)
+                    if ma is not None:
+                        return os.path.join(path, ma.group(0))
+
+        # Preprocess libraries
+        link_flags = self._get_split('Libs')
+        # Find all library paths
+        self._paths = [x for x in _list(self._re_libpath) if os.path.isdir(x)]
+        # Find all libraries
+        self.libname = _list(self._re_libname)
+        c = {}
+        # NOTE: We are making the assumption that the extension of the file is
+        # correct. This could be replaced by `os.path.realpath()` followed by
+        # calling the executable `file` to be safer.
+        location = _find_location()
+        if location:
+            ma = re.match('.*' + self._lib_pattern, location)
+            key = None
+            if ma is not None:
+                # Expected format of ma.group(1): .so or .so.1 or .a or .dylib
+                key = ma.group(1).split('.')[1]
+                c['Type'] = self._type[key]
+                extend(c, 'Location', location)
+        # Remove libraries and library paths from link flags as they are
+        # directly added to the requirements and the components.
+        link_flags = [x for x in link_flags if not x.startswith('-l')
+                      and not x.startswith('-L')]
+        extend(c, 'Link-Flags', link_flags)
+        # Add system libraries found in link flags to requirements
+        list_requires = _keys(self.requires)
+        # Remove dependency on itself.
+        # some library names already contain the prefix `lib`.
+        self.libname = [x for x in self.libname if not re.match('(lib)?' + x, self.library_name)]
+        # Update requires dictionary
+        self.requires.update({x.lower(): {} for x in self.libname if x.lower()
```

Comment:
This is a little awkward; I think it would be better to break before the `if`, rather than in the middle of the conditional expression